ws/websocket.go

```diff
@@ -316,17 +325,28 @@ func (server *Server) Start(port int, listenPath string) {
 	if server.httpServer == nil {
 		server.httpServer = &http.Server{}
 	}
+
 	addr := fmt.Sprintf(":%v", port)
 	server.httpServer.Addr = addr
 	server.httpServer.Handler = router
+
+	ln, err := net.Listen("tcp", addr)
+	if err != nil {
+		server.error(fmt.Errorf("failed to listen: %w", err))
```

Comment:
Needs a return after throwing an error, otherwise it will panic on the `ln.Addr()`, since `ln` is nil.